forage-android/src/main/java/com/joinforage/forage/android/network/data/CapturePaymentRepository.kt

```diff
@@ -151,8 +153,14 @@ internal class CapturePaymentRepository(
                 return ForageApiResponse.Failure(listOf(ForageError(500, "unknown_server_error", "Unknown Server Error")))
             }
 
+            var intervalTime: Long = if (attempt < pollingIntervals.size) {
+                pollingIntervals[attempt].toLong()
+            } else {
+                1000L
+            }
+
             attempt += 1
-            delay(POLLING_INTERVAL_IN_MILLIS + getJitterAmount())
+            delay(intervalTime + getJitterAmount())
         }
```

Comment:
the logic in this method seems almost the same (besides the log statements) as the logic in the balance polling. I've created a ticket to consolidate the polling logic into a reusable method/class to reduce this duplication.

https://linear.app/joinforage/issue/FX-610/consolidate-dedupe-polling-logic-in-android

Reply:
Thanks for making this - i'll keep that change separate from this PR actually